sdk/search/search-documents/src/searchClient.ts

```diff
@@ -164,7 +152,24 @@ export class SearchClient<T> implements IndexDocumentsClient<T> {
       apiVersion = options.apiVersion;
     }
 
-    this.client = new GeneratedClient(this.endpoint, this.indexName, apiVersion, pipeline);
+    this.client = new GeneratedClient(
+      this.endpoint,
+      this.indexName,
+      apiVersion,
+      internalClientPipelineOptions
+    );
+
+    if (isTokenCredential(credential)) {
+      this.client.pipeline.addPolicy(
+        bearerTokenAuthenticationPolicy({ credential, scopes: utils.DEFAULT_SEARCH_SCOPE })
+      );
+    } else {
+      this.client.pipeline.addPolicy(createSearchApiKeyCredentialPolicy(credential));
+    }
+
+    if (this.client.pipeline.getOrderedPolicies().length > 1) {
+      this.client.pipeline.addPolicy(createOdataMetadataPolicy("none"));
+    }
```

Comment:
Why do we need to check if the pipeline is filled? I think the previous `isArray` test was just because of how `ServiceClientOptions` was typed in `core-http`

```suggestion
    this.client.pipeline.addPolicy(createOdataMetadataPolicy("none"));
```

Reply:
Code Changed